src/main/java/org/mule/tools/devkit/sonar/utils/PomUtils.java

```diff
@@ -54,6 +60,31 @@ public static boolean isDevKitConnector(MavenProject mavenProject) {
         return parent != null && parent.getGroupId().equals(DEVKIT_GROUP_ID) && (parent.getArtifactId().equals(DEVKIT_ARTIFACT_ID) || parent.getArtifactId().equals(CERTIFIED_DEVKIT_ARTIFACT_ID));
     }
 
+    public static VersionUtils getCurrentDevkitVersion(String devkitVersion){
+        return new VersionUtils(devkitVersion);
+    }
+
+    public static VersionUtils getLatestDevkitVersion() {
+        VersionUtils latestVersion = getCurrentDevkitVersion("0.0.0");
+        try (InputStream xml = new URL("https://repository.mulesoft.org/nexus/content/repositories/releases/org/mule/tools/devkit/mule-devkit-parent/maven-metadata.xml").openStream()) {
+            Document doc = DocumentBuilderFactory.newInstance().newDocumentBuilder().parse(xml);
+            doc.getDocumentElement().normalize();
+            NodeList nodes = doc.getElementsByTagName("version");
+            latestVersion = getCurrentDevkitVersion(nodes.item(0).getFirstChild().getTextContent());
+            for (int i = 1; i < nodes.getLength(); i++) {
+                Node tag = nodes.item(i);
+                String currentValue = tag.getNodeType() == Node.ELEMENT_NODE ? tag.getFirstChild().getTextContent() : StringUtils.EMPTY;
+                // Ignore revisions (e.g. 3.7.0-M1) and Mule 4.x.x versions that refer to the new SDK
+                if (StringUtils.isNotEmpty(currentValue) && currentValue.indexOf('-') < 0  && currentValue.indexOf('4') != 0) {
+                    getCurrentDevkitVersion(currentValue).replaceIfGreaterThan(latestVersion);
+                }
+            }
+        } catch (Exception e) {
+            e.printStackTrace();
```

Comment:
Don't do `printStackTrace()`. Instead, throw the exception, wrap it inside another one, or "ignore" it if it makes sense. Just don't print it like that.